course/api/full_serializers.py

```diff
@@ -36,29 +34,35 @@
 ]
 
 
-class CourseModuleSerializer(AplusModelSerializer):
-    url = NestedHyperlinkedIdentityField(view_name='api:course-exercises-detail')
-    exercises = serializers.SerializerMethodField()
-    display_name = serializers.CharField(source='__str__')
 
-    class Meta(AplusModelSerializer.Meta):
-        model = CourseModule
-        fields = (
-            'url',
-            'html_url',
-            'display_name',
-            'is_open',
-            'exercises',
-        )
+class ModuleExerciseSerializer(serializers.Serializer):
+    id =  serializers.IntegerField()
+    url = serializers.CharField()
+    html_url = serializers.CharField()
+    display_name = serializers.CharField()
+    max_points = serializers.IntegerField()
+    max_submissions = serializers.IntegerField()
+    hierarchical_name = serializers.CharField()
+    difficulty = serializers.CharField()
 
+class CourseModuleSerializer(serializers.Serializer):
```

Comment:
Why would you change the parent class from `AplusModelSerializer` to the basic `Serializer`? Model serializers enable DRY ("don't repeat yourself") and you replaced it with manually repeated fields.